src/app/services/schema/schema_processor_service.py

```diff
@@ -105,3 +105,23 @@ def get_schema_metadata_collection_with_guid(
             schema_metadata_collection_with_guid.append(return_schema)
 
         return schema_metadata_collection_with_guid
+
+    def get_schema_bucket_filename(self, survey_id: str, version: str) -> str:
+        """
+        Gets the filename of the schema in bucket. If version is omitted,
+        the latest schema filename is retrieved
+        """
+        if version is None:
+            bucket_schema_filename = (
+                self.schema_firebase_repository.get_latest_schema_bucket_filename(
+                    survey_id
+                )
+            )
+        else:
+            bucket_schema_filename = (
+                self.schema_firebase_repository.get_schema_bucket_filename(
+                    survey_id, version
+                )
+            )
+
+        return bucket_schema_filename
```

Comment:
Sorry if this is a little nitpicky, but you don't really need to do variable assignment in here anymore, or the if/else block. So you can now do:
`
if version is None:
    return self.schema_firebase_repository.get_latest_schema_bucket_filename(survey_id)

return self.schema_firebase_repository.get_schema_bucket_filename(survey_id, version)
`

or use a ternary: 
`(
    return self.schema_firebase_repository.get_latest_schema_bucket_filename(survey_id) 
        if version is None 
        else self.schema_firebase_repository.get_schema_bucket_filename(survey_id, version)
)` 

This will cut down the amount of code you need to maintain, is still readable, and makes your code less rigid as you're no longer stuck with an if/else block